### PR TITLE
All amendments for V18.  Not yet field tested.

### LINKED ISSUE #221: ETABS_Toolkit: Support ETABSv18
#### Description:

ETABSv18 uses a new .Net framework and future versions will finally stop changing the name of their API dll every time. hopefully we can reference .NET frameworks in the same way we control which name to use.

### PR DESCRIPTION
**Pull request by RichardWhitfieldTTW**
### Issues addressed by this PR

No open issues addressed.

 Closes #221 

Functionality added to support ETABS2018.  Theoretically, this should support all versions of ETABS moving forward.

 ### Test files

No unit tests.  Project compiled hence no issue with changes to the API.

 ### Changelog

Build using either Release18 or Debug18 for a version which points to the ETABSv1.dll

 ### Additional comments

### CODE REVIEW

File: Etabs_Adapter/Read/_Read.cs [hunk 1]
```diff
@@ -20,35 +20,29 @@
...
 using BH.oM.Structure.MaterialFragments;
-#if Debug17 || Release17
+using BH.oM.Structure.SectionProperties;
+using BH.oM.Structure.SurfaceProperties;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+#if Debug18 || Release18
+#elif Debug17 || Release17
```
Comment (JosefTaylor) [inline Etabs_Adapter/Read/_Read.cs]: don't you need a using for ETABSv1 here?

File: Etabs_Adapter/Read/_Read.cs [hunk 2]
```diff
@@ -20,35 +20,29 @@
...
 using BH.oM.Structure.Loads;
 using BH.oM.Structure.MaterialFragments;
-#if Debug17 || Release17
+using BH.oM.Structure.SectionProperties;
+using BH.oM.Structure.SurfaceProperties;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+#if Debug18 || Release18
```
Comment (JosefTaylor) [inline Etabs_Adapter/Read/_Read.cs]: don't you need a using for ETABSv1 here?

File: Etabs_Adapter/Create/Bar.cs [hunk 3]
```diff
@@ -20,30 +20,26 @@
...
+using BH.Engine.Structure;
 using BH.oM.Adapters.ETABS.Elements;
 using BH.oM.Geometry.ShapeProfiles;
-#if Debug17 || Release17
-    using ETABSv17;
+using BH.oM.Structure.Elements;
+using BH.oM.Structure.MaterialFragments;
+using BH.oM.Structure.Offsets;
+using BH.oM.Structure.SectionProperties;
+#if Debug18 || Release18
```
[the remark this answers is not in the document]
Comment (alelom) [reply Etabs_Adapter/Create/Bar.cs]: I agree, something like `using ETABS` is needed here, depending on how they renamed it.

File: Etabs_Adapter/Create/Error.cs [hunk 4]
```diff
@@ -20,36 +20,24 @@
...
-using BH.oM.Structure.SurfaceProperties;
-using BH.oM.Structure.Loads;
-using BH.oM.Structure.Offsets;
-using BH.Engine.Structure;
-using BH.Engine.Geometry;
-using BH.oM.Structure.MaterialFragments;
-using BH.Engine.ETABS;
-using BH.oM.Adapters.ETABS.Elements;
-#if Debug17 || Release17
+#if Debug18 || Release18
```
Comment (alelom) [inline Etabs_Adapter/Create/Error.cs]: [Same](https://github.com/BHoM/ETABS_Toolkit/pull/242/files#r363816085) here

File: Etabs_Adapter/Create/Bar.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (alelom) [reply Etabs_Adapter/Create/Bar.cs]: I can see there is `using ETABSv1;` on other files

File: Etabs_Adapter/Create/Level.cs [hunk 5]
```diff
@@ -20,19 +20,21 @@
  * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
  */
 
+using BH.oM.Geometry.SettingOut;
 using System.Collections.Generic;
 using System.Linq;
-using BH.oM.Geometry.SettingOut;
-using BH.Engine.ETABS;
-#if Debug17 || Release17
+#if Debug18 || Release18
```
Comment (alelom) [inline Etabs_Adapter/Create/Level.cs]: [Same](https://github.com/BHoM/ETABS_Toolkit/pull/242/files#r363816085) here

File: Etabs_Adapter/Create/Link.cs [hunk 6]
```diff
@@ -20,19 +20,22 @@
...
  */
 
+using BH.oM.Structure.Constraints;
+using BH.oM.Structure.Elements;
 using System.Collections.Generic;
 using System.Linq;
-using BH.oM.Structure.Elements;
-using BH.oM.Structure.Constraints;
-#if Debug17 || Release17
+#if Debug18 || Release18
```
Comment (alelom) [inline Etabs_Adapter/Create/Link.cs]: [Same](https://github.com/BHoM/ETABS_Toolkit/pull/242/files#r363816085) here

File: Etabs_Adapter/Read/Level.cs [hunk 7]
```diff
@@ -20,22 +20,21 @@
...
-using System;
-using System.Collections;
+using BH.oM.Geometry.SettingOut;
 using System.Collections.Generic;
 using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-using BH.oM.Geometry.SettingOut;
-#if Debug17 || Release17
+#if Debug18 || Release18
```
Comment (alelom) [inline Etabs_Adapter/Read/Level.cs]: [Same](https://github.com/BHoM/ETABS_Toolkit/pull/242/files#r363816085) here

**Comment by IsakNaslundBh** [comment]
Closing this as superseded by https://github.com/BHoM/ETABS_Toolkit/pull/298

**Pull request closed without merging**